pkg/machines/hostvmslist.jsx

```diff
@@ -463,36 +467,72 @@ Vm.propTypes = {
 /**
  * List of all VMs defined on this host
  */
-const HostVmsList = ({ vms, config, dispatch }) => {
-    if (vms.length === 0) {
+class HostVmsList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.deviceProxyHandler = this.deviceProxyHandler.bind(this);
+        this.state = { hostDevices: []};
+        this.client = cockpit.dbus("org.freedesktop.NetworkManager", {});
+    }
+
+    componentWillMount() {
+        const deviceProxies = this.client.proxies("org.freedesktop.NetworkManager.Device");
```

Comment:
You only ever add and don't remove. That might cause problems. Since you loop from the names below anyways, what not just make this.proxies instead and then down in sourceJump you can loop through the proxies and find one that matches proxy.Interface. That way there is less array maintenance for you to have to handle here. Otherwsie you need to handle. added, changed and removed events.